packages/gatsby-mdx/loaders/mdx-loader.js

```diff
@@ -162,22 +162,26 @@ ${contentWithoutFrontmatter}`;
     mdPlugins: options.mdPlugins.concat(gatsbyRemarkPluginsAsMDPlugins),
     hastPlugins: options.hastPlugins
   });
-  const result = babel.transform(code, {
-    configFile: false,
-    plugins: [
-      require("@babel/plugin-syntax-jsx"),
-      require("@babel/plugin-syntax-object-rest-spread"),
-      require("../utils/babel-plugin-html-attr-to-jsx-attr")
-    ]
-  });
-  debugMore("transformed code", result.code);
-  return callback(
-    null,
-    `import React from 'react'
-import { MDXTag } from '@mdx-js/tag'
-
-
-${result.code}
-    `
-  );
+  try {
+    const result = babel.transform(code, {
+      configFile: false,
+      plugins: [
+        require("@babel/plugin-syntax-jsx"),
+        require("@babel/plugin-syntax-object-rest-spread"),
+        require("../utils/babel-plugin-html-attr-to-jsx-attr")
+      ]
+    });
+    debugMore("transformed code", result.code);
+    return callback(
+      null,
+      `import React from 'react'
+  import { MDXTag } from '@mdx-js/tag'
+
+
+  ${result.code}
+      `
+    );
+  } catch (e) {
+    callback(e);
+  }
```

Comment:
This seems safe enough, and results in Gatsby rendering the error into the browser:

![screenshot 2019-01-26 11 32 21](https://user-images.githubusercontent.com/129910/51786576-0ec5d780-215e-11e9-9089-3f0f284f1db3.png)